### PR TITLE
populate user data dynamically to page

### PR DESCRIPTION
**Pull request by Nelson-Chinedu**

### CODE REVIEW

File: components/Dashboard/Profile.tsx [hunk 1]
```diff
@@ -1,48 +1,73 @@
...
               <EdirPencil className="w1 h1 fill-white"/>
             </span>
           </div>
           <div className="ml3 ml4-ns">
-            <h4 className="mv2">Hector Johnson Okoro</h4>
-            <span className="f7">Lagos, Nigeria</span>
+            <h4 className="mv2">{data?.client?.authenticatedUser?.profile?.fullName}</h4>
+            <span className="f7">
+              {data?.client?.authenticatedUser?.profile?.city},  
+              {data?.client?.authenticatedUser?.profile?.country}
```
Comment (Hector101) [inline components/Dashboard/Profile.tsx]: do this instead
```
{city}
 {', '}
 {country}
```

File: components/Dashboard/Profile.tsx [hunk 2]
```diff
@@ -1,48 +1,73 @@
...
+    }
+  }
+}
+`;
 
 
 const Profile: FunctionComponent<{}> = () => {
+  const{ data } = useQuery(AUTHENTICATED_USER);
+
+  const bio = `${data?.client?.authenticatedUser?.profile?.bio}`;
```
Comment (Hector101) [inline components/Dashboard/Profile.tsx]: let's refactor this to 
```
const { client: { authenticatedUser: { profile: { bio, city, country } } } } = data;
```

File: components/Dashboard/Profile.tsx [hunk 3]
```diff
@@ -1,48 +1,73 @@
...
-            <span className="f7">Lagos, Nigeria</span>
+            <h4 className="mv2">{data?.client?.authenticatedUser?.profile?.fullName}</h4>
+            <span className="f7">
+              {data?.client?.authenticatedUser?.profile?.city},  
+              {data?.client?.authenticatedUser?.profile?.country}
+            </span>
+            <div>
+            <Rating
+                emptySymbol={<EmptyStar href="#icon-star-empty" className="w1 h1" />}
+                fullSymbol={<FullStar href="#icon-star-full" className="w1 h1 fill-orange" />}
```
Comment (Hector101) [inline components/Dashboard/Profile.tsx]: replace `fill-orange` with `fill-primary-blue`

File: components/Dashboard/Profile.tsx [hunk 4]
```diff
@@ -1,48 +1,73 @@
...
           </div>
           <div className="ml3 ml4-ns">
-            <h4 className="mv2">Hector Johnson Okoro</h4>
-            <span className="f7">Lagos, Nigeria</span>
+            <h4 className="mv2">{data?.client?.authenticatedUser?.profile?.fullName}</h4>
+            <span className="f7">
+              {data?.client?.authenticatedUser?.profile?.city},  
+              {data?.client?.authenticatedUser?.profile?.country}
+            </span>
+            <div>
```
Comment (Hector101) [inline components/Dashboard/Profile.tsx]: add a className attribute here and set a value `className="pv1"`

**Comment by Nelson-Chinedu** [comment]
Resolve the changes, but I have a blocker getting the rating dynamically on the page with what I tried out

**Comment by Nelson-Chinedu** [comment]
I also noticed that on the profile page, when the user details is displayed and I try to refresh the same page again I do get this error ``` TypeError: Cannot destructure property `client` of 'undefined' or 'null'. ```

File: components/Dashboard/Profile.tsx (hunk 2 above)
Comment (Hector101) [reply components/Dashboard/Profile.tsx]: this should be 
```
const aboutMe = data?.client?.authenticatedUser?.profile?.bio;
```

**Comment by Nelson-Chinedu** [comment]
I implemented the change you requested without the template literal for the ``` aboutMe ``` and got this when I still refresh the page if details is been displayed ``` TypeError: Cannot read property 'length' of undefined ```. So I had to add the template literal

**Pull request merged**